backend/opampd/opampd.go

```diff
@@ -7,19 +7,78 @@ import (
 	"net/http"
 	"strconv"
 	"sync"
+	"sync/atomic"
 	"time"
 
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/messaging"
+
 	"github.com/gorilla/mux"
 	"github.com/gorilla/websocket"
 	"github.com/open-telemetry/opamp-go/protobufs"
 	"google.golang.org/protobuf/proto"
 )
 
+const (
+	// emitMetricsEventInterval is the interval at which an OpAMP metrics event
+	// is published.
+	emitMetricsEventInterval = 30 * time.Second
+
+	statusReportsReceivedLabel = "ot-status-reports-received-total"
+	agentConfigsSentLabel      = "ot-agent-configs-sent-total"
+	errorCountLabel            = "ot-error-count-total"
+)
+
+var (
+	// statusReportsReceived is the total number of StatusReport messages
+	// received from OpenTelemetry agents.
+	statusReportReceived uint64
+
+	// agentConfigsSent in the total number of AgentRemoteConfig messages sent
+	// to OpenTelemetry agents.
+	agentConfigsSent uint64
+
+	// errorCount is the total number of errors encountered while exchanging
+	// messages with OpenTelemetry agents.
+	errorCount uint64
+)
+
+//
+func emitMetricsEvent(bus messaging.MessageBus) error {
```

Comment:
It probably is fine but do we support events without an entity?

Reply:
oh never mind, just saw sean's comment

Reply:
We don't, good catch!
Sean also mentioned it; I've just pushed a commit that uses the backend as the entity for that metrics event.